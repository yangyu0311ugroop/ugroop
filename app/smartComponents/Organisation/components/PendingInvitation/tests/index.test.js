import { DO_NOTHING } from 'appConstants';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { PendingInvitaion } from '../index';

describe('<PendingInvitaion />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    id: 999,
    classes: {},
    resaga,
  };

  beforeEach(() => {
    rendered = shallow(<PendingInvitaion {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(PendingInvitaion).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('handleClose()', () => {
    it('should not do anything', () => {
      rendered.setProps({ onClose: undefined });

      expect(instance.handleClose()).toBe(DO_NOTHING);
    });

    it('should call props.onClose', () => {
      const onClose = jest.fn();
      rendered.setProps({ onClose });

      instance.handleClose();

      expect(onClose).toBeCalledWith();
    });
  });
  describe('renderContent()', () => {
    it('should renderContent', () => {
      rendered.setProps({ ids: [] });
      // instance.renderEmptyPending = jest.fn(() => 'renderEmptyPending');
      const snapshot = shallow(<div>{instance.renderContent()}</div>);
      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render list', () => {
      rendered.setProps({ ids: ['this@guy', 'that@guy'] });

      const snapshot = shallow(<div>{instance.renderContent()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
  describe('renderHeading()', () => {
    it('should renderHeading', () => {
      const snapshot = shallow(<div>{instance.renderHeading()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });

  describe('render()', () => {
    it('should render correctly', () => {
      instance.renderHeading = jest.fn(() => 'renderHeading');
      instance.renderFilter = jest.fn(() => 'renderFilter');
      instance.renderContent = jest.fn(() => 'renderContent');

      const snapshot = shallow(<div>{instance.render()}</div>);
      expect(toJSON(snapshot)).toMatchSnapshot();
      props.showPendingOnly = true;
    });
    it('should render correctly showPendingOnly is true', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);
      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
