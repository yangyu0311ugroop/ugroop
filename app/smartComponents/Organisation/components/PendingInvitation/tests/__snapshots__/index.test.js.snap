// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PendingInvitaion /> render() should render correctly 1`] = `
<div>
  <WithStyles(Component)
    classes={
      Object {
        "paper": undefined,
      }
    }
    fullWidth={true}
    onClose={[Function]}
    open={false}
  >
    <GenericForm
      canSubmitForm={true}
      cancelButtonContent="Discard"
      disabled={false}
      discardDialogProps={Object {}}
      loading={false}
      onInvalid={[Function]}
      onValid={[Function]}
      showChangeRoutePrompt={false}
      submitButtonContent="Submit"
    >
      <div>
        <WithStyles(InlineButton)
          color="default"
          onClick={[Function]}
        >
          <WithStyles(Icon)
            icon="lnr-cross2"
          />
        </WithStyles(InlineButton)>
      </div>
      <div>
        <H1
          noMargin={true}
        >
          renderHeading
        </H1>
        <div>
          <WithStyles(GridContainer)
            direction="column"
            spacing={0}
          >
            renderContent
          </WithStyles(GridContainer)>
        </div>
        <div>
          <div />
          <H1
            noMargin={true}
          >
            <FormattedMessage
              defaultMessage="All invitations are sent via email."
              id="app.smarComponents.Organisation.PendingInvitation.footer"
            />
          </H1>
        </div>
      </div>
    </GenericForm>
  </WithStyles(Component)>
</div>
`;

exports[`<PendingInvitaion /> render() should render correctly showPendingOnly is true 1`] = `
<div>
  <WithStyles(Component)
    classes={
      Object {
        "paper": undefined,
      }
    }
    fullWidth={true}
    onClose={[Function]}
    open={false}
  >
    <GenericForm
      canSubmitForm={true}
      cancelButtonContent="Discard"
      disabled={false}
      discardDialogProps={Object {}}
      loading={false}
      onInvalid={[Function]}
      onValid={[Function]}
      showChangeRoutePrompt={false}
      submitButtonContent="Submit"
    >
      <div>
        <WithStyles(InlineButton)
          color="default"
          onClick={[Function]}
        >
          <WithStyles(Icon)
            icon="lnr-cross2"
          />
        </WithStyles(InlineButton)>
      </div>
      <div>
        <H1
          noMargin={true}
        >
          <FormattedMessage
            defaultMessage="PENDING INVITATIONS"
            id="app.smarComponents.Organisation.PendingInvitation.header"
          />
        </H1>
        <div>
          <WithStyles(GridContainer)
            direction="column"
            spacing={0}
          >
            <WithStyles(GridItem)>
              <WithStyles(GridContainer)
                direction="row"
                spacing={0}
              >
                <WithStyles(GridItem)>
                  There is no pending invitations.
                </WithStyles(GridItem)>
              </WithStyles(GridContainer)>
            </WithStyles(GridItem)>
          </WithStyles(GridContainer)>
        </div>
        <div>
          <div />
          <H1
            noMargin={true}
          >
            <FormattedMessage
              defaultMessage="All invitations are sent via email."
              id="app.smarComponents.Organisation.PendingInvitation.footer"
            />
          </H1>
        </div>
      </div>
    </GenericForm>
  </WithStyles(Component)>
</div>
`;

exports[`<PendingInvitaion /> renderContent() should render list 1`] = `
<div>
  <WithStyles(GridItem)
    key="this@guy"
  >
    <WithStyles(Connect(WithSimplifyResaga(Invitee)))
      id={999}
      index={0}
      pending={true}
      token="this@guy"
    />
  </WithStyles(GridItem)>
  <WithStyles(GridItem)
    key="that@guy"
  >
    <WithStyles(Connect(WithSimplifyResaga(Invitee)))
      id={999}
      index={1}
      pending={true}
      token="that@guy"
    />
  </WithStyles(GridItem)>
</div>
`;

exports[`<PendingInvitaion /> renderContent() should renderContent 1`] = `
<div>
  <WithStyles(GridItem)>
    <WithStyles(GridContainer)
      direction="row"
      spacing={0}
    >
      <WithStyles(GridItem)>
        There is no pending invitations.
      </WithStyles(GridItem)>
    </WithStyles(GridContainer)>
  </WithStyles(GridItem)>
</div>
`;

exports[`<PendingInvitaion /> renderHeading() should renderHeading 1`] = `
<div>
  <FormattedMessage
    defaultMessage="PENDING INVITATIONS"
    id="app.smarComponents.Organisation.PendingInvitation.header"
  />
</div>
`;
